### PR TITLE
Add option ignoreFiles to filter fileTypes

### PR DESCRIPTION
**Pull request by schoenwaldnils**
Adding option `ignoreFiles` to filter file types.
In my case this way I can filter `*.css` files to use the processor only in `js, jsx` files.

Solves #187.

### CODE REVIEW

**Comment by emilgoldsmith** [comment]
Hi, it seems you are failing some unit tests at the moment? Could you please make sure those pass, and then someone will try and get to reviewing this :)

File: src/index.js [hunk 1]
```diff
@@ -14,7 +15,8 @@ const errorWasThrown = {}
 const DEFAULT_OPTIONS = {
   moduleName: 'styled-components',
   importName: 'default',
-  strict: false
+  strict: false,
+  ignoreFiles: []
 }
 
 module.exports = options => ({
```
Comment (chinesedfan) [inline src/index.js]: Tests fail due to `options` is not normalized. A simple adjustment can be,

```
const processerFactory = options => ({
 ... ...
});
module.exports = (options) => processerFactory(Object.assign({}, DEFAULT_OPTIONS, options));
```

File: src/index.js [hunk 2]
```diff
@@ -83,25 +87,31 @@ module.exports = options => ({
...
-            lineCorrection[warning.line] || warning.line,
-            warning.column
-          )
-        })
-      )
+
+    let warnings = stylelintResult.warnings
+
+    const absolutePath = path.resolve(process.cwd(), filepath)
+    if (!micromatch(absolutePath, options.ignoreFiles).length) {
```
[the remark this answers is not in the document]
Comment (schoenwaldnils) [reply src/index.js]: Like this?
```
result(stylelintResult, filepath) {
  const err = errorWasThrown[filepath]
  const absolutePath = path.resolve(process.cwd(), filepath)
  const fileIsIgnored = micromatch(absolutePath, options.ignoreFiles).length

  if (fileIsIgnored) return stylelintResult
```

File: src/index.js (hunk 1 above)
Comment (schoenwaldnils) [reply src/index.js]: I'm not sure what you mean by the `processerFactory` function.
But I mapped the options now and the tests are green

Comment (chinesedfan) [reply src/index.js]: `processerFactory` means the whole original codes, and please forgive my typo. It should be `processorFactory` or `realProcessor`. If writing in this way, you can handle options in the unique place.

```
const realProcessor = options => ({
  code() {},
  result() {}
});
module.exports = (options) => realProcessor(Object.assign({}, DEFAULT_OPTIONS, options));
```

File: src/index.js [hunk 3]
```diff
@@ -83,7 +96,10 @@ module.exports = options => ({
     const taggedTemplateLocs = taggedTemplateLocsMap[filepath] || []
     const interpolationLines = interpolationLinesMap[filepath] || []
     const lineCorrection = sourceMapsCorrections[filepath]
-    const warnings = stylelintResult.warnings
+
+    let warnings = stylelintResult.warnings
+
+    warnings = stylelintResult.warnings
```
Comment (chinesedfan) [inline src/index.js]: Could you recover these unnecessary changes?

File: src/index.js [hunk 4]
```diff
@@ -58,6 +65,12 @@ module.exports = options => ({
   // Fix sourcemaps
   result(stylelintResult, filepath) {
     const err = errorWasThrown[filepath]
+    const mappedOptions = Object.assign({}, DEFAULT_OPTIONS, options)
+    const absolutePath = path.resolve(process.cwd(), filepath)
```
Comment (chinesedfan) [inline src/index.js]: Maybe `filepath` here is the absolute path already, or those `taggedTemplateLocs/interpolationLines` can't get value in old codes.

File: src/index.js (hunk 3 above)
Comment (schoenwaldnils) [reply src/index.js]: Done

File: src/index.js (hunk 4 above)
Comment (schoenwaldnils) [reply src/index.js]: Compared them. Were the same with the files I tested.
Therefore removed the `absolutePath`

File: src/index.js (hunk 1 above)
Comment (chinesedfan) [reply src/index.js]: @schoenwaldnils Looks like we are almost there. Would you mind processing options like above and squashing commits to less than or equal to 3? Thanks for your work in advance.

Comment (schoenwaldnils) [reply src/index.js]: Now I understood what you meant with the processing ^^
Done

**Comment by tomdev10** [comment]
Hi. Is there any documentation for this? The syntax @schoenwaldnils suggests in https://github.com/styled-components/stylelint-processor-styled-components/issues/187#issuecomment-494827256 doesn't seem to work for me. Cheers

**Comment by chinesedfan** [comment]
@tomdev10 Check #272. Notice items in config [`processors`](https://stylelint.io/user-guide/configuration#processors) are still arrays.
